vm/src/stdlib/io.rs

```diff
@@ -294,76 +292,117 @@ fn text_io_base_read(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
     }
 }
 
+fn split_mode(mode_value: String) -> Result<(String, String), String> {
+    let mut mode: char = '\0';
+    let mut typ: char = '\0';
+    let mut plus_is_set = false;
+
+    let available_modes = hashset!('a', 'r', 'w');
+    let available_types = hashset!('t', 'b');
+    for ch in mode_value.chars() {
+        if ch == '+' {
+            if plus_is_set {
+                return Err(format!("invalid mode: '{}'", mode_value));
+            }
+            plus_is_set = true;
+        } else if available_modes.contains(&ch) {
```

Comment:
Should I replace this, and next, `contains` with `if ch == 'r' && ch == 'a'...` and remove `HashSet` usage?

Reply:
That would be good, I think. You could also do `if let 'a' | 'r' | ... = c { ... }`